lib/wraft_doc_web/controllers/template_asset_controller.ex

```diff
@@ -495,8 +671,13 @@ defmodule WraftDocWeb.Api.V1.TemplateAssetController do
   end
 
   def download_public_template(conn, %{"file_name" => template_name}) do
-    with {:ok, template_url} <- TemplateAssets.download_public_template(template_name) do
+    with template_name <- ensure_zip_extension(template_name),
+         {:ok, template_url} <- TemplateAssets.download_public_template(template_name) do
       render(conn, "download_public_template.json", %{template_url: template_url})
     end
   end
+
+  defp ensure_zip_extension(name) do
+    if String.ends_with?(name, ".zip"), do: name, else: name <> ".zip"
+  end
```

Comment:
Can move this away from controller

Change 

```elixir
 @doc """
  Download template from minio.
  """
  @spec download_public_template(String.t()) :: {:ok, binary()} | {:error, String.t()}
  def download_public_template(template_name) do
    "public/templates/#{template_name}"
    |> Minio.generate_url()
    |> then(&{:ok, &1})
  end
  ```
  
 to somewhere along the line of 
 
 ```elixir
 @spec download_public_template(String.t()) :: {:ok, binary()} | {:error, String.t()}
def download_public_template(template_name) do
  case Path.extname(template_name) do
    ".zip" ->
      "public/templates/#{template_name}"
      |> Minio.generate_url()
      |> then(&{:ok, &1})
    _ ->
      {:error, "Template name must have .zip extension"}
  end
end
```